tests/test_cli.py

```diff
@@ -208,3 +242,134 @@ def mock_list(this, details=False, new=False):
+            "size": 25000,
+        },
+    ]
+
+    def mock_list(this, details=False, new=False):
+        """Mock LDP backend list method"""
+        # pylint: disable=unused-argument
+
+        response = archive_list
+        if details:
+            response = archive_list_details
+        if new:
+            response = response[1:]
+        return response
+
+    monkeypatch.setattr(FSStorage, "list", mock_list)
+
+    runner = CliRunner()
+
+    # List archives with default options
+    result = runner.invoke(cli, ["list", "-b", "fs"])
+    assert result.exit_code == 0
+    assert "\n".join(archive_list) in result.output
+
+    # List archives with detailled output
+    result = runner.invoke(cli, ["list", "-b", "fs", "-D"])
+    assert result.exit_code == 0
+    assert (
+        "\n".join(json.dumps(detail) for detail in archive_list_details)
+        in result.output
+    )
+
+    # List new archives only
+    result = runner.invoke(cli, ["list", "-b", "fs", "-n"])
+    assert result.exit_code == 0
+    assert "file2" in result.output
+    assert "file1" not in result.output
+
+    # Edge case: stream contains no archive
+    monkeypatch.setattr(FSStorage, "list", lambda this, details, new: ())
+    result = runner.invoke(cli, ["list", "-b", "fs"])
+    assert result.exit_code == 0
+    assert "Configured fs backend contains no archive" in result.output
+
+
+def test_push_command_with_fs_backend():
+    """Test the push command using the FS backend
+    It is not possible to use pyfakefs here due to a direct use of the cli which cannot be mocked
+    """
+
+    filename = "file1"
+    file_path = FS_STORAGE_DEFAULT_PATH + filename
+
+    # Create a file
+    runner = CliRunner()
+    result = runner.invoke(
+        cli,
+        [
+            "push",
+            "-b",
+            "fs",
+            "file1",
+        ],
+        input="test content",
+    )
+
+    assert result.exit_code == 0
+
+    test_file = open(file_path)
+    content = test_file.read()
+    test_file.close()
+
+    assert "test content" in content
+
+    # Try to create the same file without -f, error 1
+    runner = CliRunner()
+    result = runner.invoke(
+        cli,
+        [
+            "push",
+            "-b",
+            "fs",
+            "file1",
+        ],
+        input="different content",
+    )
+    assert result.exit_code == 1
+
+    # Try to create the same file with -f
+    runner = CliRunner()
+    result = runner.invoke(
+        cli,
+        [
+            "push",
+            "-b",
+            "fs",
+            "-f",
+            "file1",
+        ],
+        input="different content",
+    )
+
+    assert result.exit_code == 0
+
+    test_file = open(file_path)
+    content = test_file.read()
+    test_file.close()
+
+    assert "different content" in content
+
+    for archive in Path(FS_STORAGE_DEFAULT_PATH).iterdir():
+        Path(archive).unlink()
+    Path(FS_STORAGE_DEFAULT_PATH).rmdir()
```

Comment:
You don't need this while using the `fs` fixture.

Reply:
Indeed